types/webpack-dev-server/index.d.ts

```diff
@@ -353,7 +356,11 @@ declare module 'webpack' {
     }
 }
 
+type Socket = sockjs.Connection|WebSocket;
+
 declare class WebpackDevServer {
+    sockets: Socket[];
```

Comment:
for consideration, should this one be `readonly` for consumers?
```suggestion
    readonly sockets: Socket[];
```
It seems it's usually consumed like a get property, as seen here:
https://github.com/webpack/webpack-dev-server/issues/1271
```js
function reloadHtml() {
    const cache = {}
    const plugin = {name: 'CustomHtmlReloadPlugin'}
    this.hooks.compilation.tap(plugin, compilation => {
        compilation.hooks.htmlWebpackPluginAfterEmit.tap(plugin, data => {
            const orig = cache[data.outputName];
            const html = data.html.source();
            if (orig && orig !== html) {
                devServer.sockWrite(devServer.sockets, 'content-changed')
            }
            cache[data.outputName] = html;
        });
    });
}
```
```js
devServer: {
  hot: true,
  before(app, server, compiler) {
    const watchFiles = ['.html', '.hbs'];

    compiler.plugin('done', () => {
      const changedFiles = Object.keys(compiler.watchFileSystem.watcher.mtimes);

      if (
        this.hot &&
        changedFiles.some(filePath => watchFiles.includes(path.parse(filePath).ext))
      ) {
        server.sockWrite(server.sockets, 'content-changed');
      }
    });
  }
}
```

Reply:
Yes, it is mostly read only externally. I wonder if its interesting to add `readonly` as it'd break typings in the project itself, given that it pushes sockets on the `createSocketServer()` fn https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js#L703

Reply:
> I wonder if its interesting to add `readonly` as it'd break typings in the project itself, given that it pushes sockets on the

Yes, in that scenario (used as backing for webpack-dev-server itself) it would obviously. If that is a scenario for change, forget that comment